modules/ROOT/pages/schema/syntax.adoc

```diff
@@ -0,0 +1,630 @@
+a|
+[source, syntax]
+----
+CONSTRAINT [constraint_name]
+FOR ()-”[“alias [{IS \| :} IDENTIFYING_RELATIONSHIP_TYPE [=> \| IMPLIES]] “]”->()
+REQUIRE {alias.propertyName \| (alias.propertyName[, ...])} IS [REL \| RELATIONSHIP] {KEY \| UNIQUE}
+----
+|===
+
+For more information, see xref:schema/graph-types/set-graph-types.adoc#constraints-on-identifying-labels-and-types[Set graph types -> Property uniqueness and key constraints on identifying node labels and relationship types].
+
+.Syntax for property uniqueness, property existence, property type, and key constraints on non-identifying node labels and relationship types, <constraint on non-identifying label/type>
+[cols="1a, 3a", options="noheader"]
+|===
+| Node key and property uniqueness constraint
+a|
+[source, syntax]
+----
+CONSTRAINT [constraint_name]
+FOR (alias {IS \| :} NonIdentifyingLabelName)
+REQUIRE {alias.propertyName \| (alias.propertyName[, ...])} IS [NODE] {KEY \| UNIQUE}
+----
+
+| Node property existence and property type constraint
+a|
+[source, syntax]
+----
+CONSTRAINT [constraint_name]
+FOR (alias {IS \| :} NonIdentifyingLabelName)
+REQUIRE {alias.propertyName \| (alias.propertyName)} { IS NOT NULL \| {[IS] :: \| IS TYPED} <TYPE> }
+----
+
+| Relationship key and property uniqueness constraint
+a|
+[source, syntax]
+----
+CONSTRAINT [constraint_name]
+FOR ()-”[“alias {IS \| :} NON_IDENTIFYING_RELATIONSHIP_TYPE “]”->()
+REQUIRE {alias.propertyName \| (alias.propertyName[, ...])} IS [REL \| RELATIONSHIP] {KEY \| UNIQUE}
+----
+
+| Relationship property existence and property type constraint
+a|
+[source, syntax]
+----
+CONSTRAINT [constraint_name]
+FOR ()-”[“alias {IS \| :} NON_IDENTIFYING_RELATIONSHIP_TYPE “]”->()
+REQUIRE {alias.propertyName \| (alias.propertyName)} { IS NOT NULL \| {[IS] :: \| IS TYPED} <TYPE> }
+----
+|===
+
+For more information, see see xref:schema/graph-types/set-graph-types.adoc#constraints-on-non-identifying-labels-and-types[Set graph types -> Constraints on non-identifying labels and types].
+
+The three variations of the property type expression, `IS ::`, `::`, and `IS TYPED` are syntactic synonyms for the same expression.
+The preferred syntax when defining constraints in a graph type is the `IS ::` variant.
+
+`<TYPE>` can be one of the following property types:
+
+* `BOOLEAN`
+* `STRING`
+* `INTEGER`
+* `FLOAT`
+* `DATE`
+* `LOCAL TIME`
+* `ZONED TIME`
+* `LOCAL DATETIME`
+* `ZONED DATETIME`
+* `DURATION`
+* `POINT`
+* `LIST<BOOLEAN NOT NULL>`
+* `LIST<STRING NOT NULL>`
+* `LIST<INTEGER NOT NULL>`
+* `LIST<FLOAT NOT NULL>`
+* `LIST<DATE NOT NULL>`
+* `LIST<LOCAL TIME NOT NULL>`
+* `LIST<ZONED TIME NOT NULL>`
+* `LIST<LOCAL DATETIME NOT NULL>`
+* `LIST<ZONED DATETIME NOT NULL>`
+* `LIST<DURATION NOT NULL>`
+* `LIST<POINT NOT NULL>`
+* Any closed dynamic union of the above types, e.g. `INTEGER | FLOAT | STRING`.
+
+Additionally, for properties that are part of element types:
+
+* The `NOT NULL` versions of the above types
+* `ANY NOT NULL`
```

Comment:
Might be better to move up the element type specific parts to the elements section, part 2
```suggestion

```